src/components/graph/layout.ts

```diff
@@ -1,49 +1,178 @@
-    const dagreGraph = new dagre.graphlib.Graph();
-    dagreGraph.setDefaultEdgeLabel(() => ({}));
-    dagreGraph.setGraph({ align: 'UL' });
+export const nodeWidth = 230;
+export const nodeHeight = 110;
+export const snapGrid = [1, nodeHeight];
+
+type PlacementArray = string[][];
+
+function getPlacement(placementArray: PlacementArray, id: string) {
+    for (let row = 0; row < placementArray.length; row++) {
+        for (let column = 0; column < placementArray[row].length; column++) {
+            if (placementArray[row][column] === id) {
+                return { row: row, column: column };
+            }
+        }
+    }
+    return { row: -1, column: -1 };
+}
+
+function addValueAtPlacement(placementArray: PlacementArray, row: number, column: number, value: string) {
+    while (placementArray.length <= row) {
+        placementArray.push(['']);
+    }
+    while (placementArray[row].length <= column) {
+        placementArray[row].push('');
+    }
+    placementArray[row][column] = value;
+}
+
+function isSpaceEmpty(placementArray: PlacementArray, row: number, column: number) {
+    if (placementArray.length <= row) {
+        return true;
+    }
+    if (placementArray[row].length <= column) {
+        return true;
+    }
+    if (placementArray[row][column]?.length > 0) {
+        return false;
+    }
+    return true;
+}
+
+/**
+ * Builds an array representing the placements of nodes for the tree.
+ * This array is then used to compute each node's position before being used by ReactFlow.
+ */
+function getNodePlacementsFromTreeNodes(nodes: CurrentTreeNode[]) {
+    const newPlacements: PlacementArray = [];
+    let currentMaxColumn = 0;
 
     nodes.forEach((node) => {
-        dagreGraph.setNode(node.id, {
-            width: node?.type === NodeType.ROOT ? rootNodeWidth : nodeWidth,
-            height: node?.type === NodeType.ROOT ? rootNodeHeight : nodeHeight,
-        });
-    });
-    edges.forEach((edge: Edge) => {
-        dagreGraph.setEdge(edge.source, edge.target);
+        if (!node.parentId) {
+            // ROOT NODE
+            addValueAtPlacement(newPlacements, 0, 0, node.id);
+        } else {
+            // CHILDREN NODE
+            const parentPlacement = getPlacement(newPlacements, node.parentId);
+            // Check if there is an empty space below the parent
+            const tryRow = parentPlacement.row + 1;
+            const tryColumn = parentPlacement.column;
+            if (isSpaceEmpty(newPlacements, tryRow, tryColumn)) {
+                addValueAtPlacement(newPlacements, tryRow, tryColumn, node.id);
+            } else {
+                // We check if there is an empty space on the right of the used space
+                do {
+                    currentMaxColumn++;
+                } while (!isSpaceEmpty(newPlacements, tryRow, currentMaxColumn));
+                addValueAtPlacement(newPlacements, tryRow, currentMaxColumn, node.id);
+            }
```

Comment:
This will not work as intented.
Your version will produce :
```
A
┣┓
BE
┣┓
CD
```
And the solution we want is :
```
A
┣━┓
B E
┣┓
CD
```
E and D should not be on the same column.